cedar-policy-core/src/ast/policy.rs

```diff
@@ -418,6 +467,11 @@ impl Policy {
         self.template.non_head_constraints()
     }
 
+    /// Get the [`Arc`] owning non-head constraints for the policy
+    pub fn non_head_constraints_arc(&self) -> &Arc<Expr> {
+        self.template.non_head_constraints_arc()
+    }
```

Comment:
we should be careful when cherry-picking this along with #825 and #827 that this new code gets properly updated with the name changes as well